packages/confirm-dialog/src/vaadin-confirm-dialog-overlay.js

```diff
@@ -67,10 +65,28 @@ class ConfirmDialogOverlay extends DialogOverlay {
   static get template() {
     if (!memoizedTemplate) {
       memoizedTemplate = super.template.cloneNode(true);
+
+      // Replace two header slots with a single one
+      const headerPart = memoizedTemplate.content.querySelector('[part="header"]');
+      headerPart.innerHTML = '';
+      const headerSlot = document.createElement('slot');
+      headerSlot.setAttribute('name', 'header');
+      headerPart.appendChild(headerSlot);
```

Comment:
What’s the reason for this, and not using the `dialogTitle` property? Is it for backwards compatibility?

Reply:
We support setting content to `"header"` slot in the Flow counterpart - [`setHeader(Component component)`](https://github.com/vaadin/flow-components/blob/578e78de17dc46982db810af1bf49588b378d4d7/vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java#L565). Also the `"header"` slot is documented in the web component, so I don't think we can change that.

Reply:
Okay. And we don't want to keep the "title" slot available?

Reply:
I guess is might be good to expose it on the confirm dialog, too - by using the same `headerTitle` API.
Let me create a separate issue for it though, to keep this PR focused on refactoring only.